features/step_definitions/membership_application_steps.rb

```diff
@@ -48,7 +60,11 @@
       end
       ma.business_categories = categories
     end
-    ma.save
+
+    # A "legacy application" was created before we required the user to specify
+    # a file-delivery method (upload, email, etc.).  We have to skip validation
+    # for such an application since it will fail with current validation rules.
+    ma.save(validate: (legacy_app ? false : true))
```

Comment:
Won't this mean that _no_ validation will take place?  we _do_ want to make sure that the other validations happen so that we don't accidentally introduce bad data via the Background _Given these applications exist..._ tables.

Reply:
Yes it does.  I don't see a really clean way of doing this.  

I could use a specific context for the model validation(s), but that mixes in testing concerns with production code.

I could also save the app with a stand-in FDM, and then update FDM to `nil` via `update_column`.  That's a bit messy ....

So, I chose to do it this way - ensuring that the _only_ validation errors are those expected for a "legacy" app:
```
if legacy_app
  # We save without validation - so, confirm that the **only** validation errors
  # would be associated with the missing file-delivery method.
  ma.valid?
  expect(ma.errors.keys)
     .to match_array [:file_delivery_method,
                         :'user.shf_application.file_delivery_method']
end

ma.save(validate: (legacy_app ? false : true))
```

Reply:
Got it.